executor/api/rest/server.go

```diff
@@ -144,6 +155,24 @@ func (r *SeldonRestApi) Initialise() {
 	}
 }
 
+type CloudeventHeaderMiddleware struct {
+	deploymentName string
+}
+
+func (h *CloudeventHeaderMiddleware) Middleware(next http.Handler) http.Handler {
+	// TODO: Write a test that ensures the initialise function always sets deploymentName
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Should these headers only be set if the incoming request is a cloudevent otherwise it pollutes the response header space which might confuse normal REST users?

Reply:
FOr sure, I can have a look - I remember I read I read that knative eventing actually strips all the headers at the router, but I need to double check this. If it doesn't strip out the headers I can think of implementing it per request, but if it doesn't we may have to enable/disable the functionality

Reply:
I'm think to NOT have these headers for RPC use cases and only add them for eventing situations

Reply:
Oh I think my latter point was not clear - I agree with you, what I meant is that I read that the cloudevent headers are stripped and arrive as pure message, so there may not be way to know whether a requset is cloud event when it arrives. I am going to check, but if this is the case, we'll have to add a paramter to enable/disable cloudevents

Reply:
Just did a quick test, it seems that the headers are actually stripped on the router, so the executor basically doesn't really seem to get any insight on whether the request is a cloudevent. It also seems it strips other headers which is not great, perhaps there's a way to ensure knative eventing doesn't do this as that would be ideal, but in case the knative is not set up with that the only way we could deactivate it is by actually having a flag

Reply:
Ok this has been added, together with tests to check for headers to be set and not be set accordingly